docs/lib/regex.rst

```diff
@@ -0,0 +1,57 @@
+.. _regex:
+
+Scala Native Regexes
```

Comment:
I'd put on the same page as `javalib`, given that it's just a detail about or `javalib` implementation, not a separate API.